packages/vuetify/src/composables/theme.ts

```diff
@@ -0,0 +1,231 @@
+// Utilities
+import { capitalize, computed, inject, provide, ref, watch } from 'vue'
+import { useVuetify } from '@/framework'
+import { colorToInt, colorToRGB, contrastRatio } from '../util'
+
+// Types
+import type { InjectionKey, Ref } from 'vue'
+
+interface BaseColors {
+  background: string
+  surface: string
+  primary: string
+  primaryVariant: string
+  secondary: string
+  secondaryVariant: string
+  success: string
+  warning: string
+  error: string
+  info: string
+}
+
+interface OnColors {
+  onBackground: string
+  onSurface: string
+  onPrimary: string
+  onPrimaryVariant: string
+  onSecondary: string
+  onSecondaryVariant: string
+  onSuccess: string
+  onWarning: string
+  onError: string
+  onInfo: string
+}
+
+interface InternalTheme extends BaseColors, OnColors {
```

Comment:
export type IInternalTheme = BaseColors & OnColors

Or could be 

const COLORS_SLOTS = [´foo','bar'] as const
export IColorType = typeof COLORS_SLOTS[number]
export IBadeColor = Record<IColorType, string>

Could work too. They're both 1level deep Record.

(Yeah, I should do PR review from a computer)